<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab containers="1" tab_name="lqr_debug">
   <Container>
    <DockSplitter count="3" sizes="0.330463;0.339074;0.330463" orientation="-">
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-0.035214" left="1661586452.087353" top="0.031767" right="1661586457.069509"/>
        <limitY/>
        <curve color="#1f77b4" name="/lqr_debug/heading_error"/>
        <curve color="#d62728" name="/lqr_debug/lat_error"/>
        <curve color="#1ac938" name="/lqr_debug_bag/heading_error"/>
        <curve color="#ff7f0e" name="/lqr_debug_bag/lat_error"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="LinesAndDots" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-0.035555" left="1661586452.087353" top="0.045718" right="1661586457.069509"/>
        <limitY/>
        <curve color="#1ac938" name="/lqr_debug/heading_error_rate"/>
        <curve color="#ff7f0e" name="/lqr_debug/lat_error_rate"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="LinesAndDots" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-0.005226" left="1661586452.060852" top="0.006889" right="1661586457.069509"/>
        <limitY/>
        <curve color="#17becf" name="/lqr_debug/ref_curvature"/>
        <curve color="#bcbd22" name="/lqr_debug_bag/ref_curvature"/>
        <curve color="#1f77b4" name="/V2V/leader/angular_velo_z"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-16.052641" left="1661586452.060852" top="10.914623" right="1661586457.069509"/>
        <limitY/>
        <curve color="#f14cc1" name="/lqr_debug/lqr_feedback_angle"/>
        <curve color="#9467bd" name="/lqr_debug/lqr_feedforward_angle"/>
        <curve color="#d62728" name="/lqr_debug_bag/lqr_feedback_angle"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-15.891345" left="1661586452.060852" top="10.667406" right="1661586457.069509"/>
        <limitY/>
        <curve color="#ff580e" name="/lqr_debug/lqr_final_angle"/>
        <curve color="#0056a5" name="/lqr_debug/lqr_raw_angle"/>
        <curve color="#af00ff" name="/rock_can/steer_feedback/SteerAngle"/>
        <curve color="#1ac938" name="/lqr_debug_bag/lqr_final_angle"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range bottom="-0.010255" left="1661586446.376932" top="0.007643" right="1661586447.915993"/>
        <limitY/>
        <curve color="#bcbd22" name="/localization/estimation/pose/position/x"/>
        <curve color="#1f77b4" name="/localization/estimation/pose/position/y"/>
        <curve color="#d62728" name="/lqr_debug/ref_heading"/>
        <curve color="#1f77b4" name="/V2V/leader/angular_velo_z"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="dis">
   <Container>
    <DockSplitter count="1" sizes="1" orientation="-">
     <DockArea name="...">
      <plot style="LinesAndDots" mode="TimeSeries" flip_x="false" flip_y="false">
       <range bottom="0.000000" left="1661586215.852823" top="5.000000" right="1661586220.927564"/>
       <limitY min="0" max="5"/>
       <curve color="#f14cc1" name="/control/front_vehicle/curve_distance/data"/>
       <curve color="#9467bd" name="/control/front_vehicle/distance/data"/>
       <curve color="#17becf" name="/control/front_vehicle/distance_expect/data"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="0"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a ScatterXY series from arrays.&#xa;&#xa; series_name: name of the created ScatterXY series&#xa; prefix:      prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value.&#xa;             if [nil], the index of the array will be used.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{N}/position/x&#xa;   /trajectory/node.{N}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   CreateSeriesFromArray( &quot;my_trajectory&quot;, &quot;/trajectory/node&quot;,  &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;]]--&#xa;&#xa;function CreateSeriesFromArray( series_name, prefix, suffix_X, suffix_Y, timestamp )&#xa;  --- create a new series or overwite the previous one&#xa;  new_series = MutableScatterXY.new(series_name)&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_x == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

